src/dotty/tools/dotc/ast/untpd.scala

```diff
@@ -94,17 +94,51 @@ object untpd extends Trees.Instance[Untyped] with UntypedTreeInfo {
   class InfixOpBlock(leftOperand: Tree, rightOp: Tree) extends Block(leftOperand :: Nil, rightOp)
 
   // ----- Modifiers -----------------------------------------------------
+  /** Mod is intended to record syntactic information about modifiers, it's
+    * NOT a replacement of flagsets.
+    *
+    * For any query about semantic information, check `flags` instead.
+    */
+  sealed trait Mod extends Positioned
 
- /** Modifiers and annotations for definitions
-   *  @param flags          The set flags
+  object Mod {
+    case class Private() extends Mod
```

Comment:
My hunch is that this can be done slightly more elegantly as follows:

 - Have `Mod` take a `flags` field and have every case class set it. E.g.

           case class Protected() extends Mod(Flags.Protected)

Have a new method in Modifiers:

     def | (mod: Mod) =
        withFlags(mod.flags).withAddedMod(mod)
        
Then, only use `|` instead of setting flags and mods separately. I believe that would make redundant much of the current code. WDYT?